docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Elytron_OIDC_Client.adoc

```diff
@@ -0,0 +1,150 @@
+
+[standalone@localhost:9990 /] reload
+----
+
+== Configuration
+
+By default, the _elytron-oidc-client_ subsystem does not contain any configured resources or attributes.
+
+The configuration required to secure an application with OpenID Connect can either be provided within the
+application itself or within the _elytron-oidc-client_ subsystem.
+
+=== Deployment Configuration
+
+The configuration required to secure an application with OpenID Connect can be specified in the deployment.
+
+The first step is to create an `oidc.json` configuration file in the `WEB-INF` directory of the application.
+The second step is to set the `auth-method` to `OIDC` in the application's `web.xml` file.
+
+Here is an example of an `oidc.json` configuration file:
+
+[source,options="nowrap"]
+----
+{
+  "client-id" : "customer-portal",
+  "provider-url" : "http://localhost:8180/auth/realms/demo",
+  "ssl-required" : "external",
+  "use-resource-role-mappings" : false,
+  "enable-cors" : true,
+  "cors-max-age" : 1000,
+  "cors-allowed-methods" : "POST, PUT, DELETE, GET",
+  "cors-exposed-headers" : "WWW-Authenticate, My-custom-exposed-Header",
+  "enable-basic-auth" : false,
+  "expose-token" : true,
+  "verify-token-audience" : true,
+   "credentials" : {
+      "secret" : "234234-234234-234234"
+   },
+
+   "connection-pool-size" : 20,
+   "socket-timeout-millis": 5000,
+   "connection-timeout-millis": 6000,
+   "connection-ttl-millis": 500,
+   "disable-trust-manager": false,
+   "allow-any-hostname" : false,
+   "truststore" : "path/to/truststore.jks",
+   "truststore-password" : "geheim",
+   "client-keystore" : "path/to/client-keystore.jks",
+   "client-keystore-password" : "geheim",
+   "client-key-password" : "geheim",
+   "token-minimum-time-to-live" : 10,
+   "min-time-between-jwks-requests" : 10,
+   "public-key-cache-ttl": 86400,
+   "redirect-rewrite-rules" : {
+   "^/wsmaster/api/(.*)$" : "/api/$1"
+   }
+}
+----
+
+=== Subsystem Configuration
+
+Instead of adding configuration to your deployment to secure it with OpenID Connect as described in the
+previous section, another option is to add configuration to the _elytron-oidc-client_ subsystem instead.
+
+The following example shows how to add configuration to the _elytron-oidc-client_ subsystem.
+
+[source,xml,options="nowrap"]
+----
+<subsystem xmlns="urn:wildfly:elytron-oidc-client:1.0">
+    <secure-deployment name="APP_NAME.war">
+        <client-id>customer-portal</client-id>
+        <provider-url>http://localhost:8180/auth/realms/demo</provider-url>
+        <ssl-required>external</ssl-required>
+        <credential name="secret" secret="0aa31d98-e0aa-404c-b6e0-e771dba1e798" />
+    </secure-deployment
+</subsystem>
+----
+
+The `secure-deployment` resource allows you to provide configuration for a specific deployment. In
+the example above, the `secure-deployment` resource is providing the configuration that should be used
+for the `APP_NAME.war` deployment.
```

Comment:
This should be the runtime-name of the deployment, right? That is for cases where the runtime-name differs from the value of the deployment resource address / management name, the matching would be based on the runtime-name as that is what things like DUPs should deal with.

Reply:
Yes, thanks for catching that. I've updated this.